breadbox/breadbox/io/data_validation.py

```diff
@@ -56,6 +56,7 @@ def validate_all_columns_have_types(
         )
 
 
+# TODO: Replace this with annotation_type_to_pandas_column_type
```

Comment:
Sure! Right now, this mapping is used for dimension type metadata and it would be nice if we can use the more advanced pandas dtype mappings defined here in this PR but since I do not know if that will break current tests, I wanted to move this outside the scope of this PR. You're right though that I should have created a new task for this though and made the comment more clear